### PR TITLE
pyupgrade of beets to Python 3.6

### PR DESCRIPTION
**Pull request by arogl**
WIP - Attempt to upgrade to Python 3.6 as a minimum

All tests pass/fail the same as Master in both Windows (PY38) and Linux (PY38)

Pushing to get the tests to run

I used the following tool to convert https://github.com/asottile/pyupgrade
```pyupgrade --py36-plus```

## To Do

- [ ] Documentation. (If you've add a new command-line flag, for example, find the appropriate page under `docs/` to describe it.)
- [ ] Changelog. (Add an entry to `docs/changelog.rst` near the top of the document.)
- [ ] Tests. (Encouraged but not strictly required.)

### CODE REVIEW

**Comment by arogl** [comment]
How do you want me to pull in the updates to Master?
per commit or by Pull rrquest

**Comment by sampsyo** [comment]
Ah, great question—sorry for letting this fall out of date! Any option is fine with me, TBH (merging in master here, rebasing, etc.)—all options seem perfectly good to me.

File: beets/autotag/match.py [hunk 1]
```diff
@@ -108,7 +106,7 @@ def assign_items(items, tracks):
     log.debug('...done.')
 
     # Produce the output matching.
-    mapping = dict((items[i], tracks[j]) for (i, j) in matching)
+    mapping = {items[i]: tracks[j] for (i, j) in matching}
```
Comment (sampsyo) [inline beets/autotag/match.py:109]: This is a clever improvement. Nice work, pyupgrade.

File: beets/autotag/mb.py [hunk 2]
```diff
@@ -15,28 +14,23 @@
...
+from collections import Counter
+from urllib.parse import urljoin
 
 VARIOUS_ARTISTS_ID = '89ad4ac3-39f7-470e-963a-56509c546377'
 
-if util.SNI_SUPPORTED:
-    BASE_URL = 'https://musicbrainz.org/'
-else:
-    BASE_URL = 'http://musicbrainz.org/'
+BASE_URL = 'https://musicbrainz.org/'
```
Comment (sampsyo) [inline beets/autotag/mb.py:33]: Really great that we can get rid of this check.

File: beets/dbcore/db.py [hunk 3]
```diff
@@ -748,8 +746,8 @@ def _make_model(self, row, flex_values={}):
         """ Create a Model object for the given row
         """
         cols = dict(row)
-        values = dict((k, v) for (k, v) in cols.items()
-                      if not k[:4] == 'flex')
+        values = {k: v for (k, v) in cols.items()
+                  if not k[:4] == 'flex'}
```
Comment (sampsyo) [inline beets/dbcore/db.py:750]: Could be `k.endswith('flex')`, I suppose.

File: beets/importer.py [hunk 4]
```diff
@@ -1317,17 +1317,16 @@ def read_tasks(session):
 
         # Generate tasks.
         task_factory = ImportTaskFactory(toppath, session)
-        for t in task_factory.tasks():
-            yield t
+        yield from task_factory.tasks()
```
Comment (sampsyo) [inline beets/importer.py:1320]: Wahoo! One of my favorite Python 3 quality-of-life syntax features.

File: beets/library.py [hunk 5]
```diff
@@ -451,85 +451,85 @@ class Item(LibModel):
     _table = 'items'
     _flex_table = 'item_attributes'
     _fields = {
-        'id':       types.PRIMARY_ID,
-        'path':     PathType(),
+        'id': types.PRIMARY_ID,
+        'path': PathType(),
         'album_id': types.FOREIGN_ID,
 
-        'title':                types.STRING,
```
Comment (sampsyo) [inline beets/library.py:458]: It's a little sad to lose this table-like formatting, which did make this listing more readable, but I suppose it's not that big of a deal.

File: beetsplug/mpdstats.py [hunk 6]
```diff
@@ -60,13 +55,8 @@ def __init__(self, log):
...
         self._log.debug('strip_path: {0}', self.strip_path)
 
-        if sys.version_info < (3, 0):
-            # On Python 2, use_unicode will enable the utf-8 mode for
-            # python-mpd2
-            self.client = mpd.MPDClient(use_unicode=True)
-        else:
-            # On Python 3, python-mpd2 always uses Unicode
-            self.client = mpd.MPDClient()
+        # On Python 3, python-mpd2 always uses Unicode
```
Comment (sampsyo) [inline beetsplug/mpdstats.py:58]: This comment is maybe not necessary anymore.

File: setup.cfg [hunk 7]
```diff
@@ -1,5 +1,5 @@
 [flake8]
-min-version = 2.7
+min-version = 3.6
```
Comment (sampsyo) [inline setup.cfg:2]: Nice catch; I left this one out.

File: setup.py [hunk 8]
```diff
@@ -87,7 +85,6 @@ def build_manpages():
     },
 
     install_requires=[
-        'six>=1.9',
```
Comment (sampsyo) [inline setup.py:90]: :tada: :tada: :tada:

File: test/rsrc/convert_stub.py [hunk 9]
```diff
@@ -30,9 +25,9 @@ def convert(in_file, out_file, tag):
 
     # On Windows, use Unicode paths. On Python 3, we get the actual,
     # Unicode filenames. On Python 2, we get them as UTF-8 byes.
-    if platform.system() == 'Windows' and PY2:
-        in_file = in_file.decode('utf-8')
-        out_file = out_file.decode('utf-8')
+    # if platform.system() == 'Windows' and PY2:
+    #    in_file = in_file.decode('utf-8')
+    #    out_file = out_file.decode('utf-8')
```
Comment (sampsyo) [inline test/rsrc/convert_stub.py:30]: Probably OK to delete?

**Review by sampsyo — APPROVED** [review]
Wow! This is an enormous diff, but I think I've now reviewed the whole thing. I only found a few very minor/superficial comments. Aside from that, I think we should merge this as soon as possible! So excited to be rid of the dependency on `six`.

File: beetsplug/mpdstats.py (hunk 6 above)
Comment (arogl) [reply beetsplug/mpdstats.py]: Deleted

File: test/rsrc/convert_stub.py (hunk 9 above)
Comment (arogl) [reply test/rsrc/convert_stub.py]: deleted

File: beets/library.py (hunk 5 above)
Comment (arogl) [reply beets/library.py:458]: Happy to reformat

**Comment by arogl** [comment]
@sampsyo I believe I have addressed the feedback

**Comment by sampsyo** [comment]
Truly heroic, @arogl. This is so awesome.

**Pull request merged**